### PR TITLE
[GEP-18] Prepare reusing secrets manager in extensions

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area security
/kind enhancement
/merge squash

**What this PR does / why we need it**:

Prepare reusing secrets manager in extensions by
- documenting what the consider
- introducing secrets manager identity
- augmenting relevant test cases

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/3292#issuecomment-1036058890

**Special notes for your reviewer**:

Note that `user-kubeconfig` and `kube-apiserver-static-token` secrets will leak if you rotate the kubeconfig of shoots created with `v1.43.0-dev` before reconciling them with this revision. As this is only relevant for shoot created with an unreleased version and the feature is in alpha state, we don't bother adding migration logic.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/development/secrets_management.md [hunk 1]
```diff
@@ -69,6 +71,20 @@ if err != nil {
 
 As explained above, this returns the bundle secret for the CA `my-ca` which might potentially contain both the current and the old CA (in case of rotation/regeneration).
 
+## Reusing the SecretsManager in Other Components
```
Comment (rfranzke) [inline docs/development/secrets_management.md:74]: Can you somehow link to this section from `docs/extensions`? I haven't checked which place might be appropriate, though.

Comment (timebertt) [reply docs/development/secrets_management.md:74]: Good suggestion, done :)

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**